project-base/storefront/helpers/pagination/loadMore.ts

```diff
@@ -122,23 +130,205 @@ export const getRedirectWithOffsetPage = (
     }
 
     const updatedQuery: ParsedUrlQuery = getFilteredQueries(currentQuery);
+    const searchParams = new URLSearchParams();
+
+    for (const [key, value] of Object.entries(updatedQuery)) {
+        if (!value) {
+            continue;
+        }
+
+        if (Array.isArray(value)) {
+            value.forEach((v) => searchParams.append(key, v));
+        } else {
+            searchParams.set(key, value);
+        }
+    }
```

Comment:
Niiice! I guess much better than use legacy library or an another library in general for such a small case. Thanks for your effort 🙏